### PR TITLE
RavenDB-18543 Missing HasRevisions flag when importing docs and revision independently

### PR DESCRIPTION
**Pull request by YonatanResh**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18543/Missing-HasRevisions-flag-when-importing-docs-and-revision-independently

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Is it platform specific issue?

- No

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works
- It has been verified by manual testing

### CODE REVIEW

File: src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs [hunk 1]
```diff
@@ -1461,6 +1462,11 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                         {
                             _database.DocumentsStorage.RevisionsStorage.Put(context, id, document.Data, document.Flags,
                                 document.NonPersistentFlags, document.ChangeVector, document.LastModified.Ticks);
+
+                            if (addHasRevisionsFlag.Contains(id) == false && document.Flags.Contain(DocumentFlags.HasRevisions))
```
Comment (grisha-kotler) [inline src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs]: This isn't the document but the revision.
You need to check if the actual document is missing the `Revisions` flag.

**Review by grisha-kotler — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/DocumentsStorage.cs [hunk 2]
```diff
@@ -1549,9 +1549,11 @@ private static Document ParseDocumentPartial(JsonOperationContext context, ref T
             if (fields.Contain(DocumentFields.ChangeVector))
                 result.ChangeVector = TableValueToChangeVector(context, (int)DocumentsTable.ChangeVector, ref tvr);
 
+            if (fields.Contain(DocumentFields.Flags))
```
Comment (YonatanResh) [inline src/Raven.Server/Documents/DocumentsStorage.cs:1552]: This is probably a breaking change, we wanted to see which tests will fail to help us see where we need to fix

File: src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs (hunk 1 above)
Comment (YonatanResh) [reply src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs]: done

File: src/Raven.Server/Documents/DocumentsStorage.cs (hunk 2 above)
Comment (YonatanResh) [reply src/Raven.Server/Documents/DocumentsStorage.cs:1552]: I couldn't find anything else that needed to be changed becuse of this

**Review by grisha-kotler — APPROVED** [review]

**Comment by karmeli87** [comment]
Will be handled by @shaharhikri after https://issues.hibernatingrhinos.com/issue/RavenDB-20425 is done

**Pull request closed without merging**